amd_tressfx_vulkan/src/TressFXRendererVulkan.cpp

```diff
@@ -0,0 +1,1569 @@
+        vr = (x);                                                                        \
+        if (vr != VK_SUCCESS)                                                            \
+        {                                                                                \
+            return vr;                                                                   \
+        }                                                                                \
+    }
+#endif
+
+// unreferenced formal parameter
+#pragma warning(disable : 4100)
+
+using namespace DirectX;
+
+extern int g_TressFXNumVerticesPerStrand;
+
+struct CB_PER_FRAME
+{
+    XMMATRIX m_mWorld;
+    XMMATRIX m_mViewProj;
+    XMMATRIX m_mInvViewProj;
+    XMMATRIX m_mViewProjLight;
+
+    XMFLOAT3 m_vEye;
+    float m_fvFOV;
+
+    XMFLOAT4 m_AmbientLightColor;
+    XMFLOAT4 m_PointLightColor;
+    XMFLOAT4 m_PointLightPos;
+    XMFLOAT4 m_MatBaseColor;
+    XMFLOAT4 m_MatKValue;
+
+    float m_FiberAlpha;
+    float m_HairSMAlpha;
+    float m_bExpandPixels;
+    float m_FiberRadius;
+
+    XMFLOAT4 m_WinSize;
+
+    float m_FiberSpacing;
+    float m_bThinTip;
+    float m_fNearLight;
+    float m_fFarLight;
+
+    int m_iTechSM;
+    int m_bUseCoverage;
+    int m_iStrandCopies;
+    int m_iMaxFragments;
+
+    float m_alphaThreshold;
+    float m_fHairKs2;
+    float m_fHairEx2;
+    unsigned m_optionalSRVs;
+
+    XMMATRIX m_mInvViewProjViewport;
+
+    int m_mNumVerticesPerStrand;
+    int m_mNumFollowHairsPerGuideHair;
+    int m_bSingleHeadTransform;
+    int padding0;
+};
+
+// Optional SRVs
+#define PER_STRAND_TEX_COORDS 0x0001
+#define PER_VERTEX_TEX_COORDS 0x0002
+#define PER_VERTEX_COLORS 0x0004
+
+struct PER_PIXEL_LINKED_LIST_STRUCT
+{
+    DWORD TangentAndCoverage;
+    DWORD depth;
+    DWORD strandColor;
+    DWORD dwNext;
+};
+
+struct PPLL_BUFFERS
+{
+    // Buffers for the head of the per-pixel linked lists (PPLL)
+    VkImage pHeadPPLL_Buffer;
+    VkDeviceMemory pHeadPPLL_Memory;
+    VkImageView pHeadPPLL_View;
+
+    // Buffers for the per-pixel linked list (PPLL) data
+    VkBuffer pPPLL_Buffer;
+    VkDeviceMemory pPPLL_Memory;
+
+    // Buffer for the atomic counter
+    VkBuffer pAtomicCounterPLL_Buffer;
+    VkDeviceMemory pAtomicCounterPLL_Memory;
+
+    int width;
+    int height;
+    int refCount; // reference count - delete buffers when 0
+};
+
+PPLL_BUFFERS g_PPLBuffers = {nullptr, nullptr, nullptr, 0, 0, 0};
+
+const static UINT g_HairTotalLayers = 32;
+
+namespace
+{
+//----------------------------------------------------------------------------------
+//
+// CreateRenderPass
+//
+// Create the render pass used for the non shortcut algorithm
+//
+//----------------------------------------------------------------------------------
+VkRenderPass CreateRenderPass(VkDevice dev)
+{
+    const VkAttachmentDescription attachments[] = {
+        // Depth Buffer
+        {0, VK_FORMAT_D24_UNORM_S8_UINT, VK_SAMPLE_COUNT_1_BIT,
```

Comment:
Don't use `VK_FORMAT_D24_UNORM_S8_UINT` which is not supported by AMD hardware. Query the format support instead and prefer a D32 (without stencil) if possible.